caffe2/python/operator_test/adagrad_test.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import functools
+
+from hypothesis import given
+import numpy as np
+
+from caffe2.python import core
+import caffe2.python.hypothesis_test_util as hu
+
+
+class TestAdagrad(hu.HypothesisTestCase):
+
+    @staticmethod
+    def ref_adagrad(param_in, mom_in, grad, lr, epsilon):
+        mom_out = mom_in + np.square(grad)
+        grad_adj = lr * grad / (np.sqrt(mom_out) + epsilon)
+        param_out = param_in + grad_adj
+        return (param_out, mom_out)
+
+    @given(inputs=hu.tensors(n=3), **hu.gcs)
+    def test_adagrad(self, inputs, gc, dc):
+        # Inputs
+        param, momentum, grad = inputs
+        lr = np.random.rand(1).astype(np.float32)
+
+        # Op parameters
+        epsilon = np.random.rand(1).astype(np.float32)[0]
```

Comment:
Please make them arguments of the function with hypothesis inputs. Using randomness inside the test confuses hypothesis and makes it run forever